js-exercises/animal-sort/animalSort.js

```diff
@@ -0,0 +1,32 @@
+const animalSort = animals => {
+  if (!(animals instanceof Array)) {
+    throw Error(`Expected Array as an input, got ${typeof animals}`);
+  }
+
+  const requiredKeys = ['numberOfLegs', 'name'];
+
+  for (const animal of animals) {
+    if (!(requiredKeys.every(key => Object.keys(animal).includes(key)))) {
+      throw Error('Expected keys : "numberOfLegs" and "name" not found.');
+    }
+  }
+
+  if (animals.length < 1) {
+    return [];
+  }
```

Comment:
It's repeat mistake of https://github.com/pesto-students/deep-dive-1akshat/pull/2#discussion_r414694818 which was resolved